### ISSUE TITLE
[BUG/ISSUE] dryrun not finding existing files in 12.9.0+

### ISSUE THREAD

**Issue by ltmurray** [geoschem/geos-chem#379]
### Expected behavior: ###
Executing ./geos --dryrun should successfully find existing files on the filesystem

### Actual behavior: ###
Executing ./geos --dryrun in version 12.9.0 or 12.9.1 fails to find most existing files on the filesystem, including meteorology, but does find volcanoes, PARANOX and some CHEM_INPUTS

### Steps to reproduce: the bug:
1. git clone a fresh build of 12.9.1 from github
2. cp -rpd a fresh geosfp_4x5_standard run directory from the Unit Tester
3. Execute ./geos --dryrun
[dryrun.log](https://github.com/geoschem/geos-chem/files/4940021/dryrun.log)

**Compilation commands**
cd /path/to/rundir
mkdir build && cd build && cmake path/to/Code.12.9.1
make -j install

**Run commands**
./geos --dryrun &> dryrun.log

### Error messages
All of the files in the dryrun log file, except the GFED4 2020 directory, exist on my system

In dryrun.log, for example:
/scratch/ltmurray_lab/data/input/gc//ExtData/GEOS_4x5/GEOS_FP/2016/07/GEOSFP.20160701.A1.4x5.nc
HEMCO: REQUIRED FILE NOT FOUND 

But on the system
[ltmurray@bhc0003]$ if [ -f /scratch/ltmurray_lab/data/input/gc//ExtData/GEOS_4x5/GEOS_FP/2016/07/GEOSFP.20160701.A1.4x5.nc ]; then echo Found; fi
Found

### Your GEOS-Chem version and runtime environment:
 - GEOS-Chem version: 12.9.1
 - Compiler version: ifort 17.0.5
 - netCDF version: 4.7.3
 - netCDF-Fortran version (if applicable): 4.5.2
 - Did you run on a computational cluster, on the AWS cloud: no
 - Are you using GEOS-Chem "out of the box" (i.e. unmodified): yes

### Input and log files to attach

 - lastbuild: __
 - input.geos: __
 - HEMCO_Config.rc: __
 - GEOS-Chem "Classic" log file: attached as dryrun.log
 - HEMCO.log: __
 - slurm.out or any other error messages from your scheduler: __
 - Any other error messages: __

**Comment by ltmurray** [comment]
I found the problem.  When SrcFile_Parse is called during dryrun, FOUND is not being set to HasFile, and therefore returns its initial value (.FALSE.).  In the IF statement below, we just need to set FOUND to HasFile before the RETURN.

```
    IF ( HcoState%Options%isDryRun ) THEN

       ! Make sure that the year is not 1, this indicates that the
       ! preferred year is outside of the years specified in the
       ! time range settings in the configuration file, and will
       ! lead to files with a year of "0001" in the path.
       ! (bmy, 6/9/20)
       IF ( prefyr == 1 ) THEN
          MSG = 'Cannot find file for current simulation time: ' // &
               TRIM(srcFile) // ' - Cannot get field ' // &
               TRIM(Lct%Dct%cName) // '. Please check file name ' // &
               'and time (incl. time range flag) in the config. file'
          CALL HCO_ERROR( HcoState%Config%Err, MSG, RC )
          RETURN
       ENDIF

       ! Otherwise return with success
       RC = HCO_SUCCESS
       **FOUND = HasFile**
       RETURN
    ENDIF
```

**Comment by lizziel** [comment]
Thanks Lee!

**Comment by yantosca** [comment]
We will add this fix into 12.9.2.

**Comment by yantosca** [comment]
I have just added the fix that @ltmurray provided above as commit https://github.com/geoschem/geos-chem/commit/650b66c3de4308b5c848f51f9870ab5ccfc7b7d8.  This should now resolve the issue.  Thanks for the fix!
